internal/backends/postgresql/metadata/registry.go

```diff
@@ -104,3 +248,432 @@ func (r *Registry) DatabaseList(ctx context.Context) ([]string, error) {
 
 	return res, nil
 }
+
+// DatabaseGetExisting returns a connection to existing database or nil if it doesn't exist.
+//
+// If the user is not authenticated, it returns error.
+func (r *Registry) DatabaseGetExisting(ctx context.Context, dbName string) (*pgxpool.Pool, error) {
+	defer observability.FuncCall(ctx)()
+
+	p, err := r.getPool(ctx)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	r.rw.RLock()
+	defer r.rw.RUnlock()
+
+	db := r.colls[dbName]
+	if db == nil {
+		return nil, nil
+	}
+
+	return p, nil
+}
+
+// DatabaseGetOrCreate returns a connection to existing database or newly created database.
+//
+// The dbName must be a validated database name.
+//
+// If the user is not authenticated, it returns error.
```

Comment:
Do I understand correctly that it's because `getPool` returns an error?
Should we add the same comment to the `DatabaseList` method doc? I see all the methods that call `getPool` have this comment, but not `DatabaseList`.

Reply:
Yes I'll add it.